### ISSUE TITLE
Sprint Feb 5 to Feb 16, 2024

### ISSUE THREAD

**Issue by mariusandra** [PostHog/posthog#20055]
# Global Sprint Planning

## 3 things that might take us down

1. Django (sync) can only take a certain number of concurrent requests. 150 should be enough for anyone? We should upgrade.
2. 
3.

# Team sprint planning

For your team sprint planning copy this template into a comment below for each team.

```
# Team ___

**Support hero:** ___

## Retro

<!-- Grab the high and low priority items from last time and add whether that item was completed or not -->

- 

## Hang over items from previous sprint

<!-- For each item, decide to re-prioritise (and add below) or deprioritise -->

- Item 1. prioritised/deprioritise

## OKR

1. OKR, status (red/yellow/green) and action points if yellow/red


### High priority

-

### Low priority / side quests

-

```

**Comment by mariusandra** [comment]
# Team Product Analytics

**Support hero:** 
- Julian

## Retro

- 👍 HogQL rewrite (get trends, stickiness, lifecycle, retention out) @Gilbert09 @webjunkie @mariusandra 
  - ✅ Lifecycle - out
  - ✅ Retention - out
  - 🔲 Stickiness - might be out by now?
  - 🔲 Trends - largely finished, needs thorough testing
  - 🟦 Paths - mostly finished, waiting for funnels for last bit
  - 🟦 Funnels - started
- 👍 Start funnels @thmsobrmlr 
- 👍 Support our users @Twixes 
- 😶 Django upgrade
- 👍 Multiple celery consumers

## OKR Q1 2024

1. HogQL & Data Exploration (Julian, Marius)
   - Convert ALL our insights to use HogQL as their base. **ongoing**
   - Improve the type system and get rid of `assumeNotNull`.
   - Write great docs for HogQL and data exploration nodes. **ongoing**
   - Proactively monitor query performance.

2. Product Analytics frontend (Thomas, Michael)
   - Clean things up now that we have PostHog 3000 and HogQL everywhere.
   - Work on the universal "Explore" view. **ongoing**

3. BI (Tom)
   - Support non-event data sources on HogQL insights.
   - Build the new querying experience, and/or integrate it into the "Explore" view.

<details>
<summary>Action plan</summary>
- Come January, we'll take [this list](https://github.com/PostHog/meta/issues/130), split the checkboxes between the team, and keep going till it's done. ETA: end of January.
- In February (leap year, woohoo), we'll make Product Analytics ~great~ spark joy again: insights UI cleanup, unified data exploration experience, adding missing analysis features (lifecycle breakdowns, etc), and in parallel work on the BI and the HogQL editing experience. What _exactly_ we'll do and in what order is still TBD.
- In March we'll keep going. The goal is to have an amazing BI / Product Analytics / Insights / Explore / HogQL experience by the end of it.
</details>

## Next sprint

### High priority

Rough first draft. To be discussed at standup.

- HogQL insights rewrite (trends, paths, funnels)
- HogQL features (autocomplete, type system, JSON support)
- Pick up the BI work again

### Low priority / side quests

- Django upgrade

**Comment by robbie-c** [comment]
TODO

**Comment by pauldambra** [comment]
Team Replay

Support hero: @pauldambra 

everybody is in the whole sprint 🚀 

## Retro
- still pretty hard to get feedback on session recording on android... only really had feedback from two or three that have volunteered @marandaneto 
- grabbing example apps and testing it there ourselves - @pauldambra 
- canvas stuff is super exciting @pauldambra 
- should we open source it already @marandaneto 

## Hang over items from previous sprint

✅ network payload capture released for all @pauldambra
🟡 2 users using Android replay @marandaneto (we did get two testers)
✅ BETA canvas recording for replay @daibhin 
🟡 AI session summary @pauldambra 
✅ ~~(Wrap up comments work) - @benjackwhite and @daibhin needs some follow-up.~~
✅ POC of session replay with jetpack compose @marandaneto (can't do wireframe for jetpack compose)
✅ @daibhin take over this PR to add resizeable inspector tab for the player https://github.com/PostHog/posthog/pull/18378
🟡 initial POC/discovery on React Native recordings - complicated, started, but slow going
 
## OKR

- 🟡📱 happy android replay users
  - feedback from beta testers but not a working recording from them yet 😓 
- 🟡 people love replay even more
  - still more bug reports than we'd like
- 🟡 ish  first class mobile support
  - not focussed here, focus is on replay
  - but making small pieces of progress
  - fixing code snippets - better RN and flutter support

## High priority

- using embeddings/clustering to generate session summaries @daibhin @pauldambra 
    - generating clusters
    - naming them
    - not aiming to productise but validate that we can generate magic playlists 
-  start iOS recording so we can pull from larger tester pool @marandaneto 
- 2 more Android testers @pauldambra @marandaneto 

## Side quests

-  contribute the network capture back to rrweb - so they don't make one that doesn't work for us - @pauldambra 
- do we release comments/discussion @daibhin 
- POC RN recordings @marandaneto 
- mobile recording documentation update @marandaneto @pauldambra 
- OMG mobile dashboards @pauldambra (@marandaneto)
- buffering/loading/performance @pauldambra 

from growth review

- network payload capture on onboarding?
- adding billing limits to unsubsscribe?

**Comment by benjackwhite** [comment]
# Team Infra

**Support hero:** @danielxnj 

## Retro



## Hang over items from previous sprint

1. SOC2 compliance work mostly done
2. Amazon procurement stuff

1. Rollout new ingestion endpoints and static asset accessing @frankh 
  - ✅ Most of the legwork is done here - now its just about carefully rolling out
3. Scoped investigation into async django to relieve the concurrent requests issue @frankh @benjackwhite 
  - ❌ no progress here
  - Could be that django upgrades help us here so lets focus on reliable canary deploys
5. Scoped investigation into Soc2 (type I or II) with Fraser @danielxnj 
  - ✅ mostly done now
  - In addition Amazon procurement process stuff
1. Test out one of our Canary Deploy ideas
  - ✅ Tested gradual rollouts on dev using Contour / Argo

## OKR

1. 💪 Deploy with confidence
1. 💰 Keep costs under control
1. 👀 More Observability
1. 🎯 Soc2

### High priority

1. Keep rolling out the new ingestion ingresses with client-side support @frankh @benjackwhite 
  - why? It is delicate and almost there - let's focus on getting it done
3. Move the final work into `charts` (posthog in particular) @danielxnj 
  - why? This is chore work but blocks some of the canary deployments and is still a lot of mental overhead.

### Low priority / side quests
- Get changed CH deployment work into Terraform @danielxnj

**Comment by raquelmsmith** [comment]
# Team Growth

## Retro
<details>
<summary>Retro items</summary>

- [x] We shouldn’t show pricing above $20k/yr on the pricing pages so we don’t “shoot ourselves in the foot” for larger customers who will be paying above that - @xrdt 
  - Could be an experiment, does it put people off from signing up because it isn’t as transparent, or feels enterprise-y. 
- [x] Make doc for how to create the right kinds of prices in Stripe @raquelmsmith 
- [x] Experiment: redirect people to web analytics after onboarding instead of insights screen - @xrdt 
- [x] Hide replay controls for people on the old plan - @xrdt 
- [x] New product intro pages - https://github.com/PostHog/posthog/pull/19609 - @raquelmsmith 
- [x] Wrap up finished experiments - remove code, close experiment @xrdt 
  - [x] invite members https://us.posthog.com/project/2/experiments/8176
  - [x] post-ingestion redirect to insights https://us.posthog.com/project/2/experiments/8865
- [ ] Product intro pages @raquelmsmith 
</details>


## OKR
- 🟡 Double activation rates for product analytics and session replay
  - ⚪ Polish onboarding flow with Cory
  - 🟡 Run 1 experiment per week
  - 🟡 Do an average of 2 customer interviews per week
- ⚪ Launch pricing for Data warehouse, queries, CDP


## This sprint
### High priority
- [x] Make new celery queue for usage reports
- [ ] Full testing of the onboarding flow @raquelmsmith - snapshots done, e2e pending
- [ ] Fix mrr per product for credit-based annual customers - @raquelmsmith 
- [ ] Feature flag to turn off quota limiting during billing migrations @xrdt 
  - Basically done https://github.com/PostHog/posthog/pull/19996
- [ ] Don't drop data for 7 (or maybe 3) days until after it's done
  - [ ] Pending on @tiina303 as she has an interesting idea for how pipeline can handle this in plugin-server.
    - [ ] Pipeline team won't handle until Q2 so we'll handle for the time being. 
    - Do we need this? Is the feature flag enough? Most people complain after they've noticed it's a problem already, so a delay wouldn't necessarily help, and the billing migration thing we can take care of with the flag.
    - But I'd hate to say we didn't do this AGAIN if it would have fixed something...
- [ ] Onboarding reskin based on Cory's mocks @xrdt 
- [ ] Teams plan @raquelmsmith 
- [ ] Come up with 5 ideas to bump the session replay activation rate @raquelmsmith 
- [ ] Rethink some activation metrics (remove team member invited, double check rates with diff activities) @raquelmsmith - in progress


### Low priority / side quests
- [ ] Write two articles for Andy
- [ ] Expired sessions should redirect users to the login page rather than popping lots of error toasts.
## Growth review items
- [x] Add metadata to startup plan customers in stripe @raquelmsmith

**Comment by EDsCODE** [comment]
# Team Data Warehouse

### OKR Q1 2024

### Objective

**Get EU moved over to new cluster**

* **Key Results:**
    * Terraform, ansible, data migrated over, backups, housewatch
    * Deprecate old cluster

**ByConity evaluation**

* **Key Results:**
    * Load up with all clickhouse data and benchmark

**5 reference customers with qualities:**

* **Key Results:**
    * Paying a reasonable amount
    * Tell us they’re delighted
    * Not a big edge case
    * Using our in house data import pipeline
    * Weekly queries on a data warehouse connected table
    * Saving queried tables to insights?

**Data Warehouse queries are compatible with insights hogQL**

 

### Retro

**Product**

- [x] postgres integration
- [x] Side tracked with looking into flaky data import jobs due to dropped postgres connection
- [ ] data warehouse table as first class data in insights


**Infra** [Board](https://github.com/orgs/PostHog/projects/92/views/1)
- [ ] Ensure non-sharded backups backup only once
- [ ] Finish incremental backing up of large tables
- [ ] Ansible configs for all EU CH nodes. Core, Coordinator, ZK

### High priority

**Product**
- [ ] stop flaky jobs
- [ ] test postgres integration live
- [ ] experiment with adding imported data to persons table

**Infra**

**Comment by neilkakkar** [comment]
# Team Feature Success

**Support hero:** @liyiy (surveys) + @jurajmajerik   
Days off:
Juraj: 2 days
Neil: 1 days
Li: 1 days

## Retro

- e2e tests for experiments - @jurajmajerik - basic e2e tests there, will expand when we add more functionality & work on existing functionality again
- Experiment bug fixes, common sense boundaries for predictions, update predictions after updating metrics - @jurajmajerik -> some fixes & validation for events is in, common sense boundaries deferred till after UI revamp which will change a lot of this now.
-  Statistical significance on secondary metrics & display secondary metrics charts - @neilkakkar -> Will roll over, rolling out relative date ops took longer than expected, some decide errors with cloudflare, focusing on fixing that first.
- Finish converting surveys to preact, and use it for previews in posthog app - @liyiy -> conversion will get done! Previews to follow after things are merged
- Start on event based targeting implementation (RFC here: https://github.com/PostHog/product-internal/pull/556/ )- @liyiy - will roll 😋 


## Hang over items from previous sprint

---

## OKRs

1. Doubling down on surveys
2. Doubling down on experiments

### High priority

- Experiments UI create & view overhaul - @jurajmajerik
- Fix control is never significant in experiments - @jurajmajerik 
- Survey results are squished fix; delayed surveys; nicer results loading screens; more results filtering in surveys - @liyiy
- Statistical significance on secondary metrics & display secondary metrics charts - @neilkakkar 
- Refactor FeatureFlagReleaseConditions to work well with scheduled changes, surveys, and flags - @neilkakkar 


### Low priority / side quests

- Temporal queues for feature success - @neilkakkar 

---

We'll do - Event based triggers for surveys (RFC here: https://github.com/PostHog/product-internal/pull/556/ )- @liyiy  - after this sprint.
